firebase-functions/firebase.js

```diff
@@ -0,0 +1,20 @@
+const config = {
+  apiKey: process.env.GATSBY_FIREBASE_API_KEY,
+  authDomain: process.env.GATSBY_FIREBASE_AUTH_DOMAIN,
+  databaseURL: process.env.GATSBY_FIREBASE_DATABASE_URL,
+  projectId: process.env.GATSBY_FIREBASE_PROJECT_ID,
+  storageBucket: process.env.GATSBY_FIREBASE_STORAGE_BUCKET,
+  messagingSenderId: process.env.GATSBY_FIREBASE_MESSAGING_SENDER_ID,
+};
+
+let firebaseInstance;
+export const getFirebase = firebase => {
```

Comment:
can you point me to the reference you got this pattern from? I want to understand it better and why we need it

Reply:
I followed this to make it work: https://kyleshevlin.com/firebase-and-gatsby-together-at-last

Reply:
ok, thanks, that makes sense!